MentorMatchmaker/MentorMatchmaker/MatchMakingDomainOperations.fs

```diff
@@ -248,12 +293,62 @@ module Matchmaking =
                 }
         )
 
+        (* Dump unmatched applications *)
+        
         let unmatchedApplicants = transformedMenteesInUnmatchedApplicants @ transformedMentorsInUnmatchedApplicants
 
         let fileContent = 
             unmatchedApplicants 
             |> List.map(fun application -> $"{dumpToFileApplicationData application}")
             |> String.concat("\n")
 
-        System.IO.File.WriteAllText("applicationDataDump.txt", fileContent)
+        System.IO.File.WriteAllText("applicationDataDumpUnmatchedApplications.txt", fileContent)
+
+        (* Dump unmatched applications -- Pairing permutations *)
+        
+        let unmatchedApplicantsPairings = createUnmatchedApplicantPairingPermutations transformedMenteesInUnmatchedApplicants transformedMentorsInUnmatchedApplicants
+
+        let renderMentorPairingPermutations (permutations: {| Mentees: {| Mentee: UnmatchedApplicant; Mentor: UnmatchedApplicant; SessionHours: OverlapSchedule list |} list; Mentor: UnmatchedApplicant |}) =
+            let renderTopics applicant = 
+                let interests = applicant.Interests |> NonEmptyList.map(fun x -> x.Category.CategoryName) |> String.concat(", ")
+                $"Topics: {interests}\n"
+
+            let dumpMeetingTimes (meetingTimes: OverlapSchedule nel) =
+                meetingTimes
+                |> NonEmptyList.map(fun meetingDay ->
+                    let aggregatedTimes = meetingDay.MatchedAvailablePeriods |> NonEmptyList.toList |> List.fold(fun accumulatedTimes currentTime -> accumulatedTimes + $", {currentTime.UtcStartTime}") ""
+                    let aggregatedTimes = aggregatedTimes.Substring(2)
```

Comment:
What is the 2 for?

Reply:
Sorry, was a while since I wrote this and didn't notice this when I made the PR. I reviewed again, and turns out the original 'generateMeetingTimes' isn't safe. Especially the opration `NonEmptyList.create mentorWeekSchedule.Head mentorWeekSchedule.Tail` can fail when then `mentroWeekSchedule` is empty (and thus has no head). This probably wasn't an issue in the original code since the list would never be empty but for the permutations this could happen. I quickly created the second version to not mess with the original.

I think we could probably just replace the original with my modified version. Or is there a reason why we'd work with the NonEmptyList internally?

The other comments of duplicated code would be resolved with this as well, since I just copied the original codeblock.